novawallet/Modules/AssetsSearch/AssetOperation/Send/SendAssetsOperationInteractor.swift

```diff
@@ -0,0 +1,82 @@
+import BigInt
+
+final class SendAssetsOperationInteractor {
+    weak var presenter: AssetsSearchInteractorOutputProtocol?
+
+    let stateObservable: AssetListStateObservable
+    let filter: ChainAssetsFilter
+    let logger: LoggerProtocol
+
+    private var builder: AssetSearchBuilder?
+
+    init(
+        stateObservable: AssetListStateObservable,
+        logger: LoggerProtocol
+    ) {
+        self.stateObservable = stateObservable
+        self.logger = logger
+
+        filter = { chainAsset in
+            let assetMapper = CustomAssetMapper(type: chainAsset.asset.type, typeExtras: chainAsset.asset.typeExtras)
+
+            guard let transfersEnabled = try? assetMapper.transfersEnabled(), transfersEnabled else {
+                return false
+            }
+            guard let balance = try? stateObservable.state.value.balanceResults[chainAsset.chainAssetId]?.get() else {
+                return false
+            }
+
+            return balance > 0
+        }
+    }
+}
+
+extension SendAssetsOperationInteractor: AssetsSearchInteractorInputProtocol {
+    func setup() {
+        let operationQueue = OperationQueue()
+        operationQueue.maxConcurrentOperationCount = 1
+
+        builder = .init(
+            filter: filter,
+            state: createState(from: stateObservable.state.value),
+            workingQueue: .main,
+            callbackQueue: .main,
+            callbackClosure: { [weak self] result in
+                self?.presenter?.didReceive(result: result)
+            },
+            operationQueue: operationQueue,
+            logger: logger
+        )
+
+        builder?.apply(query: "")
+
+        stateObservable.addObserver(with: self) { [weak self] _, newState in
+            guard let self = self else {
+                return
+            }
+            self.builder?.apply(state: self.createState(from: newState.value))
```

Comment:
I think we must separate builder internal model from the state model. Currently it is hard to track the logic when we take a model from the state modify it and put to another builder that actually doesn't need part of it. Meaning, we passing around balances only for the purpose to model the same state model and pass to the builder.